File: java/leetcode/src/main/java/com/leetcode/TwoPointers/Long_Pressed_Name_925.java
package com.leetcode.TwoPointers;

public class Long_Pressed_Name_925 {
    public boolean isLongPressedName(String name, String typed) {
        char pre = ' ';
        int i = 0;
        int j = 0;
        for (; j < typed.length() && i < name.length();) {
            char a = name.charAt(i);
            char b = typed.charAt(j);
            if (i == 0) {
                if (a != b) return false;
                else {
                    i++;
                    j++;
                    pre = a;
                }
            } else {
                if (a == b) {
                    i++;
                    j++;
                    pre = a;
                } else {
                    if(b == pre) {
                        j++;
                    }else {
                        return false;
                    }
                }
            }
        }

        for (; j < typed.length();) {
            char b = typed.charAt(j);
            if (b == pre) j++;
            else return false;
        }

        return i == name.length() && j == typed.length();

    }

    public static void main(String[] agrs) {
        Long_Pressed_Name_925 test = new Long_Pressed_Name_925();
        String name = "vtkgn";
        String typed = "vttkgnn";

        boolean res = test.isLongPressedName(name, typed);
        System.out.println(res);
    }
}

File: js/algrithm/two-pointer/Long_Pressed_Name_925.js
/**
 * @param {string} name
 * @param {string} typed
 * @return {boolean}
 */
var isLongPressedName = function(name, typed) {
    const arr1 = name.split('')
    const arr2 = typed.split('')
    const m = arr1.length
    const n  = arr2.length
    let i = 0
    for (let j = 0; j < n; j++) {
        if (i < m && arr1[i] === arr2[j]) {
            i++
        } else {
            if (i === 0) return false
            if (arr1[i - 1] !== arr2[j]) return false
        }
    }
    return i === m
};

const name = "alex", typed = "aaleexa"
const res = isLongPressedName(name, typed)
console.log(res)